src/globus_sdk/services/auth/client/confidential_client.py

```diff
@@ -294,3 +294,106 @@ def oauth2_token_introspect(
+            linebreaks.
+        :type name: str
+        :param public_client: This is used to infer which OAuth grant_types the client
+            will be able to use. Should be false if the client is capable of keeping
+            secret credentials (such as clients running on a server) and true if it is
+            not (such as native apps). After creation this value is immutable.
+        :type public_client: bool
+        :param visibility: If set to "public", any authenticated entity can view it.
+            When set to "private", only entities in the same project as the client can
+            view it.
+        :type visibility: str
+        :param redirect_uris: list of URIs that may be used in OAuth authorization
+            flows.
+        :type redirect_uris: iterable of str, optional
+        :param terms_and_conditions: URL of client's terms and conditions.
+        :type terms_and_conditions: str, optional
+        :param privacy_policy: URL of client's privacy policy.
+        :type privacy_policy: str, optional
+        :param required_idp: In order to use this client a user must have an identity
+            from this IdP in their identity set.
+        :type required_idp: str or uuid, optional
+        :param preselect_idp: This pre-selects the given IdP on the Globus Auth login
+            page if the user is not already authenticated.
+        :type preselect_idp: str or uuid, optional
+        :param additional_fields: Any additional parameters to be passed through.
+        :type additional_fields: dict, optional
+
+
+        .. tab-set::
+
+            .. tab-item:: Example Usage
+
+                .. code-block:: pycon
+
+                    >>> ac = globus_sdk.AuthClient(...)
+                    >>> project_id = ...
+                    >>> r = ac.create_child_client(
+                    ...     "My client",
+                    ...     True,
+                    ...     True,
+                    ... )
+                    >>> client_id = r["client"]["id"]
+
+            .. tab-item:: Example Response Data
+
+                .. expandtestfixture:: auth.create_child_client
+
+            .. tab-item:: API Info
+
+                ``POST /v2/api/clients``
+
+                .. extdoclink:: Create Client
+                    :ref: auth/reference/#create_client
+        """
+        body: dict[str, t.Any] = {
+            "name": name,
+            "public_client": public_client,
+        }
+        if visibility is not None:
+            body["visibility"] = visibility
+        if redirect_uris is not None:
+            body["redirect_uris"] = list(utils.safe_strseq_iter(redirect_uris))
+        if required_idp is not None:
+            body["required_idp"] = str(required_idp)
+        if preselect_idp is not None:
+            body["preselect_idp"] = str(preselect_idp)
+
+        # terms_and_conditions and privacy_policy must both be set or unset
+        if type(terms_and_conditions) is not type(privacy_policy):
```

Comment:
Very minor, but I wonder if we consider this cleaner if, rather than `type(x) is not type(y)`, we use `bool` + X-or?
```
bool(x) ^ bool(y)
```

It's pretty similar, so I can be convinced, but I thought I should mention it.

Reply:
Using `bool` is fine. I think this was a holdover from `update_client()` where using `terms_and_conditions=None, privacy_policy=MISSING` would have passed `bool` but failed in Auth.

FWIW, I'm not sure if the SDK should check these anyways (1. Auth could remove the requirement, 2. SDK is supposed to be a thin layer). But I saw similar minor checks like this in other calls so I added it. I think those mostly dealt with query params though.

Reply:
fixed in upcoming commit

Reply:
Regarding safety checking in the client:
I'm fine with checks like these. I don't want us to do anything elaborate or expensive client-side, but if it's easy to add a guard like this against a "trap" where it looks like you can do something with an API which we know is invalid, I'm in favor.

We can always remove these guards _if_ there's a problem/bug with them which we can't trivially fix, because we aren't declaring it as part of the API surface -- function signatures are much tougher to modify later.